File: lab1-js/lab_js_oop/rectangle.js
import {Figure} from './figure.js'
import {Color} from './color.js'

export class Rectangle extends Figure {
    constructor(width, height, color) {
        super()
        this.type = 'Прямоугольник'
        this.width = width
        this.height = height
        this.color = new Color(color)
    }

    area() {
        return this.width * this.height
    }

    repr() {
        return `${this.type} ${this.color.color} цвета с длиной стороны ${this.width}, шириной стороны ${this.height}, площадью ${this.area()}`
    }
}

File: lab1-test/lab_python_oop/rectangle.py
from .figure import  Figure
from .color import Color


class Rectangle(Figure):
    type = "Прямоугольник"

    def __init__(self, width, height, color):
        self.width = width
        self.height = height
        self.color = Color(color)

    def area(self):
        return self.width * self.height

    def __repr__(self):
        return '{} {} цвета с длиной стороны {}, шириной стороны {}, площадью {}.'.format(
            self.type,
            self.color.val,
            self.width,
            self.height,
            self.area()
        )

